src/Npgsql.Json.NET/Internal/JsonNetTypeInfoResolverFactory.cs

```diff
@@ -26,7 +26,7 @@ static TypeInfoMappingCollection AddMappings(TypeInfoMappingCollection mappings,
             {
                 var jsonb = dataTypeName == "jsonb";
                 mappings.AddType<JToken>(dataTypeName, (options, mapping, _) =>
-                    mapping.CreateInfo(options, new JsonNetJsonConverter<JToken>(jsonb, options.TextEncoding, settings)), isDefault: true);
+                    mapping.CreateInfo(options, new JsonNetJsonConverter<JToken>(jsonb, options.TextEncoding, settings)));
```

Comment:
I'm sure I'm missing context here, but don't we *want* JToken to be mapped to `jsonb` by default? That makes more sense to me as the default mapping...

Reply:
Maybe, but then there are a few problems:

1. This is a breaking change (and seems to be unintentional) from 7.0 and below, where we always mapped to string by default.
2. We do not do the same for STJ, which ideally should mirror Json.NET.

Reply:
> This is a breaking change (and seems to be unintentional) from 7.0 and below, where we always mapped to string by default.

OK, but given that 8.0 has been out for a year, wouldn't changing back break everyone already relying on the new behavior?

> We do not do the same for STJ, which ideally should mirror Json.NET.

I agree that we should ideally mirror JSON.NET, but I'm not sure that's a strong enough reason here... At the point where we are, changing Json.NET back to string would be a breaking change that's similar to changing STJ to jsonb; if we think jsonb is the better mapping (and I tend to lean in that direction), shouldn't we rather consider going the other way?

Reply:
I'm probably missing something, but how jsonb is relevant here? The issue is, do we want to map both json and jsonb to a string by default while reading even if user enables either STJ or Json.NET, or we return some abstract value (like JToken).
While I do think the second option (returning an object) is probably better, currently not only we have a mismatch between STJ and Json.NET (where STJ returns a string while Json.NET JToken), but we also changed the previous behavior in a patch version (8.0.2, although for 8.0.0 we've been trying to return JToken which seems to be not compatible with JArray). So IMO for now we should return to the previous behavior and officially change it for 10.0.

Reply:
Wait, I think I misunderstood - I thought this is about writing, i.e. what PG type should be sent by default when the user writes JToken (PG jsonb or text).

If this is about reading, i.e. what CLR type should be returned when reading a PG `text` (.NET string or JToken), then I'm definitely fine to always return string by default, regardless of which JSON plugin is set up.

However, If we do end up returning to the previous behavior and then "officially" change it again later, I think that would do our users a disservice with constant breaks back and forth... I think it's better to decide what we really want the behavior to be long-term and decide where to go from there...

Reply:
> Wait, I think I misunderstood - I thought this is about writing, i.e. what PG type should be sent by default when the user writes JToken (PG jsonb or text).

Yeah, that's just reading, writing shouldn't be affected (at least from my understanding).

> However, If we do end up returning to the previous behavior and then "officially" change it again later, I think that would do our users a disservice with constant breaks back and forth... I think it's better to decide what we really want the behavior to be long-term and decide where to go from there...

I'm mostly OK with whatever. If a user reads json, they're most likely using POCO, at which case they'll call `GetValue<T>`, in turn whatever is returned by default doesn't affect them. All in all string seems to be the safest choice, as it doesn't make it suddenly return another type of value depending on whether you enabled json or not.

Reply:
Yeah, agreed!